timemachine/md/minimizer.py

```diff
@@ -488,18 +488,10 @@ def local_minimize(
     x_local_shape = (n_local, 3)
     u_0, _ = val_and_grad_fn(x0)
 
-    # deal with overflow, empirically obtained by testing on some real systems.
-    guard_threshold = 5e4
-
     def val_and_grad_fn_local(x_local):
         x_prime = x0.copy()
         x_prime[local_idxs] = x_local
         u_full, grad_full = val_and_grad_fn(x_prime)
-        # avoid being trapped when overflows spuriously appear as large negative numbers
-        # remove after resolution of https://github.com/proteneer/timemachine/issues/481
-        if not np.isfinite(u_full) or u_0 - u_full > guard_threshold:
```

Comment:
Is there still a circumstance where we'd want to convert NaNs to inf here?

Reply:
I don't remember if the fixed point code now can still emit nans (since iirc every fixed point energy now has a representable float32 representation).

Reply:
It can still return Nan. https://github.com/proteneer/timemachine/blob/master/tests/test_energy_overflows.py#L23 but has to be a pretty egregious clash. 

I think it could help to convert the NaN to an inf, will add back the `if not np.isfinite(u_full)`

Reply:
Added back in the `np.isfinite(u_full)` in https://github.com/proteneer/timemachine/pull/1364/commits/d9940a78c33ecaad68e0a9122b2e4a00d273b4c1

Though do we want this to catch `-np.inf` in the case of using the jax platform?

Reply:
Maybe `np.isnan`, specifically to handle the case where the GPU platform encounters a fixed-point overflow and returns NaN?

Reply:
Would we want to set the forces to Nan when the energies are +/- inf?

Reply:
> Would we want to set the forces to Nan when the energies are +/- inf?

I might be missing something, but I don't see why this is necessary.

Reply:
Added `isnan` check in https://github.com/proteneer/timemachine/pull/1364/commits/2fad3ce6e8425c84f28d42bb41ba41090ab4a98e

> I might be missing something, but I don't see why this is necessary.

Just don't know BFGS well enough to know what the behavior is when the energy is non-finite while the gradients are finite.